app/com/linkedin/drelephant/tunin/FitnessComputeUtil.java

```diff
@@ -0,0 +1,231 @@
+                .eq(TuningJobDefinition.TABLE.job + "." + JobDefinition.TABLE.id, job.id)
+                .eq(TuningJobDefinition.TABLE.tuningEnabled, 1).findUnique();
+
+        logger.debug("Job Execution Update: Flow Execution ID " + jobExecution.flowExecution.flowExecId + " Job ID "
+            + jobExecution.jobExecId);
+
+        List<AppResult> results =
+            AppResult.find
+                .select("*")
+                .fetch(AppResult.TABLE.APP_HEURISTIC_RESULTS, "*")
+                .fetch(
+                    AppResult.TABLE.APP_HEURISTIC_RESULTS + "." + AppHeuristicResult.TABLE.APP_HEURISTIC_RESULT_DETAILS,
+                    "*").where().eq(AppResult.TABLE.FLOW_EXEC_ID, jobExecution.flowExecution.flowExecId)
+                .eq(AppResult.TABLE.JOB_EXEC_ID, jobExecution.jobExecId).findList();
+
+        if (results != null && results.size() > 0) {
+          Long totalExecutionTime = 0L;
+          Double totalResourceUsed = 0D;
+          Double totalInputBytesInBytes = 0D;
+
+          for (AppResult appResult : results) {
+            logger.info("Job Execution Update: ApplicationID " + appResult.id);
+            //            Long executionTime = appResult.finishTime - appResult.startTime - appResult.totalDelay;
+            //            totalExecutionTime += executionTime;
+            totalResourceUsed += appResult.resourceUsed;
+            totalInputBytesInBytes += getTotalInputBytes(appResult);
+          }
+
+          Long totalRunTime = Utils.getTotalRuntime(results);
+          Long totalDelay = Utils.getTotalWaittime(results);
+          totalExecutionTime = totalRunTime - totalDelay;
+
+          if (totalExecutionTime != 0) {
+            jobExecution.executionTime = totalExecutionTime * 1.0 / (1000 * 60);
+            jobExecution.resourceUsage = totalResourceUsed * 1.0 / (1024 * 3600);
+            jobExecution.inputSizeInBytes = totalInputBytesInBytes;
+            logger.info("Job Execution Update: UpdatedValue " + totalExecutionTime + ":" + totalResourceUsed + ":"
+                + totalInputBytesInBytes);
+          }
+
+          logger.debug("Job execution " + jobExecution.resourceUsage);
+          logger.debug("Job details: AvgResourceUsage " + tuningJobDefinition.averageResourceUsage
+              + ", allowedMaxResourceUsagePercent: " + tuningJobDefinition.allowedMaxResourceUsagePercent);
+
+          //Compute fitness
+          if (jobExecution.executionState.equals(JobExecution.ExecutionState.FAILED)
+              || jobExecution.executionState.equals(JobExecution.ExecutionState.CANCELLED)) {
+            // Todo: Check if the reason of failure is auto tuning and  handle cancelled cases
+            tuningJobExecution.fitness =
+                3 * tuningJobDefinition.averageResourceUsage * tuningJobDefinition.allowedMaxResourceUsagePercent
+                    * 1024.0 * 1024.0 * 1024 / (100.0 * tuningJobDefinition.averageInputSizeInBytes);
```

Comment:
Replace all these constants and use FileUtils library. Here and elsewhere

Reply:
Done.